checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java

```diff
@@ -1,5 +1,7 @@
 package org.checkerframework.checker.guieffect;
 
+import static org.checkerframework.framework.type.AnnotatedTypeFactory.ParameterizedMethodType;
```

Comment:
It's a bit strange that this is statically imported

Reply:
Overall, replacing the `Pair` with a dedicated class is definitely good.
I also don't like the static imports. Are they needed because `ParameterizedMethodType` is an inner class?

Reply:
> Are they needed because ParameterizedMethodType is an inner class?

Yes.  The alternative is writing `AnnotatedTypeFactory.ParameterizedMethodType` everywhere that `ParameterizedMethodType` is currently used.

Reply:
I thought static imports where for importing methods and fields -- not classes.  The code compiles even when static is removed.

Reply:
I agree with @smillst . The `static` isn't needed and the code would be nicer without.